otherlibs/dune-rpc/private/procedures.ml

```diff
@@ -9,14 +9,115 @@ module Public = struct
   end
 
   module Diagnostics = struct
+    module V1 = struct
```

Comment:
V1 of the diagnostics API is just the previous version sharing as much with the new as possible. Basically anywhere `unit Pp.t` is mentioned, we use the lossy conversions to get them back. On the other hand it will mean that new dunes listening to old clients won't have colour support, which is fine I think.

Reply:
Could you move this to a separate file? E.g. `diagnostics_v1.ml`?